woocommerce/woocommerce-android
# Order creation single product selection

Summary
==========
Partially fixes issue #5277 by creating the Product selection view with the product list basics.

Since this PR serves as the base branch for #5573 and a lot of improvements for this PR are introduced there, I'll leave the remaining improvements to be tackled in an upcoming PR to avoid conflicts with that PR and cause rework.

Screenshots
==========
| Order Creation form with no selected product  | Product selection list after clicking on `Add Product` | Order Creation form after a Product is selected |
| ------------- | ------------- | ------------- |
| ![Screenshot_20220103_203252](https://user-images.githubusercontent.com/5920403/147991599-05230f70-015a-4af7-a8f5-71b2a8f39a3a.png) | ![Screenshot_20220103_203306](https://user-images.githubusercontent.com/5920403/147991605-ef0c8fe9-7069-40f7-86f5-0cdcbe635289.png) | ![Screenshot_20220103_203314](https://user-images.githubusercontent.com/5920403/147991610-298eb4fd-b68c-445e-8d10-ba4f0158f7f3.png) | 

How to Test
==========
1. Make sure the order creation beta flag is enabled.
2. Open Order's list, and click on the FAB button.
3. Click on Add Product.
4. Verify that the Add Product view is presented with the correct Product list from the given store (excluding all Variable products)
5. Click on any Product and verify if the view returns to the Order Creation form with the expected product added


Update release notes:
